crates/sparrow-runtime/src/prepare/preparer.rs

```diff
@@ -185,6 +183,53 @@ impl Preparer {
             self.time_multiplier.as_ref(),
         )
     }
+
+    /// Creates the output url prefix to use for prepared files.
+    ///
+    /// e.g. for osx: file:///<dir>/<KASKADA_PATH>/tables/<table_uuid>/prepared/<uuid>
+    ///      for s3:  s3://<path>/<KASKADA_PATH>/tables/<table_uuid>/prepared/<uuid>
+    pub fn prepared_output_url_prefix(
```

Comment:
It feels like we're over complicating this. What if we just made the prefix optional -- if it is `None` then we write to the temp dir, if it is `Some(URL)` then the URL must be a directory where we will append the `tables/<table_uuid>/prepared/<uuid>` for preparation purposes.

That seems like it handles the cases:

1. Prepare to the temp dir. `None`.
2. Prepare to a specific local folder. `Some("file://path/to/prepare/in/")`.
3. Prepare to a specific object store folder. `Some("s3://path/to/prepare/in/")`.

Reply:
We could, but then we're always passing both the `tempdir` and the `option<url>` everywhere

Reply:
Why? We create the tempdir in the session, and it lives long enough. When you create the session, you take an `prepare_path_prefix: Option<Url>`. You then convert it to: `prepare_path_prefix: ObjectStoreUrl`. Every where else, you just pass `&prepare_path_prefix`:

```
let tempdir = tempdir::NamedTempDir::new();
let prepare_path_prefix = prepare_path_prefix.unwrap_or(|| {
  ObjectStoreUrl::from_filesystem_dir(tempdir.path())
});
Ok(Session { tempdir, prepare_path_prefix })
```

Now we've let the user configure the prepare path prefix, but we only need to pass that everywhere else.